addons/print/models/print_printer.py

```diff
@@ -195,70 +248,66 @@ def spool_report(self, docids, report_name, data=None, title=None, copies=1):
                 reports = self.env.ref(name, raise_if_not_found=False)
             if not reports:
                 raise UserError(_("Undefined report %s") % name)
-
-        # Identify required report types
-        report_types = set(reports.mapped("report_type"))
-
-        # If there is only 1 report type to print out then raise an error if printer not found,
-        # otherwise continue checking for other report types
-        printer_raise_if_not_found = len(report_types) == 1
-
-        # Identify printer to use for each report type, only include report type in dictionary
-        # if a printer is identified
+        return reports
+
+    def _assign_printers_to_report(self, required_report_types):
+        """
+        Assign each printer in self (or system default if self is empty) to a given
+        report type.
+        Return: dict{ ir.action.report() : print.printer() }
+        """
+        printer_raise_if_not_found = len(required_report_types) == 1
```

Comment:
Q: Do we only raise an error for printer not being found if there are multiple report types because we might have be able to print out some of the types but not others?

Might be worth adding a comment or something in the doc-string.

Reply:
On closer inspection I realised that if we always pass True then _check_every_report_type_has_printer() is redundant. 

_check_every_report_type_has_printer() would check that each report type has a printer, but if we just raise the error when no printer is found for a report type - which we can do in printers() - we do not need _check_every_report_type_has_printer().